backend/apps/medical_catalog/services/hospital_item_price_service.py

```
@@ -12,23 +15,48 @@ def create(*, data: dict, user=None) -> HospitalItemPrice:
+            # Check if this is a uniqueness validation error, otherwise re-raise
+            if hasattr(e, 'message_dict'):
+                for field, messages in e.message_dict.items():
+                    if any('already exists' in str(msg).lower() for msg in messages):
+                        raise DuplicateError("HospitalItemPrice", [field], f"HospitalItemPrice with this {field} already exists")
+            # Not a uniqueness error - re-raise original ValidationError
+            raise
+        except IntegrityError as e:
+            # Database constraint violation
+            error_msg = str(e).lower()
+            if 'unique' in error_msg:
+                raise DuplicateError("HospitalItemPrice", message="HospitalItemPrice with these values already exists")
```

Ruff (0.14.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)